radis/io/hitran.py

```diff
@@ -1409,6 +1466,29 @@ def fetch_hitran(
     # Delete files if needed:
     if cache == "regen":
         ldb.remove_local_files(local_file)
+    else:
+        # Raising AccuracyWarning if local_file exists and doesn't have extra columns in it
+        if ldb.get_existing_files(local_file):
+            import vaex
```

Comment:
That's the right thing to do, unfortunately we do not use Vaex all the time. It's the new default but on some systems we still rely on Pandas, and it may change in the future.
That's why Radis includes a Hdf5Manager class, to abstractize the file format / reading library 

In your case I think you'll need to write a get_columns() function that will work with at least Vaex and Pandas 

For Vaex its exactly what you wrote 
For pandas it's more tricky. If you open the database it will actually load it in memory, thus take a lot of time and memory. But you can read columns without loading everything. See for instance : https://stackoverflow.com/questions/49401464/hdfstore-get-column-names/56403217#56403217

If format is not Pandas or Vaex just read a NotImplementeferror

Implementation : 

Get_columns could be a method of the DatabaseManager class (parent of HITRAN Database Manager above).
Use : 
```
Ldb.get_columns()
```
Docstring : returns list of columns in the database without reading the full database in memory . If many files, ensure that all files have the same columns names, or raise an error.




It's a new part of the codebase for you so ask questions if needed

Reply:
done